### PR TITLE
Add ICount demultiplex-annotate-analyse workflow

### PR DESCRIPTION
**Pull request by jkokosar**

### CODE REVIEW

**Comment by mstajdohar** [comment]
@JureZmrzlikar, please review if the fields are correct. Check name, label, type and value choices where applicable. @jkokosar, rebase on to master and log changes.

**Comment by JureZmrzlikar** [comment]
I really had no contact with the experiment (annotation) procedure. Therefore, I don't feel competent to judge weather fields are correct.

**Comment by jkokosar** [comment]
@mstajdohar I have added the changelog and rebased onto master. This PR should remain as WIP until it is reviewed by @jinlajf

**Comment by mstajdohar** [comment]
Rebase and fix tests.

**Comment by jkokosar** [comment]
@mstajdohar @tjanez I have expanded an existing PR, since additions were required also for the original commit. We now need to add the ability for tests to connect to the localhost before I can expand the tests to properly validate the workflow functionality.

File: resolwe_bio/processes/icount/annotate_samples.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- slug: icount-demultiplex-annotate
+  name: iCount demultiplex and annotate
+  description: Annote iCount samples after demultiplexing
```
Comment (mstajdohar) [inline resolwe_bio/processes/icount/annotate_samples.yml]: I would change this to Demultiplex and annotate iCount samples.

File: resolwe_bio/tools/icount_demultiplex_annotate.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Run iCount demultiplexing.")
+    parser.add_argument('annotation_file', help="iCount sample annotation file.")
+    parser.add_argument('annotation_id', type=int, help="Annotation file object ID")
+    parser.add_argument('reads_id', type=int, help="Reads file object ID")
+    parser.add_argument('data_id', type=int, help="Current data ID")
+    return parser.parse_args()
+
+
+def _tsv_to_dict(table_file):
```
Comment (mstajdohar) [inline resolwe_bio/tools/icount_demultiplex_annotate.py]: The same function in two files. Can you import the function from another module?

**Pull request merged**